app/src/main/java/com/example/nasaapp/data/repository/ArticleDetailsNetworkDataSource.kt

```diff
@@ -0,0 +1,42 @@
+package com.example.nasaapp.data.repository
+
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.nasaapp.data.api.TheArticleDBInterface
+import com.example.nasaapp.data.value_object.ArticleDetails
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+
+
+class ArticleDetailsNetworkDataSource (private val apiService : TheArticleDBInterface, private val compositeDisposable: CompositeDisposable) {
+    private val _networkState = MutableLiveData<NetworkState>()
+    val networkState : LiveData<NetworkState>
+        get() = _networkState
+
+    private val _downloadedArticleDetailsResponse = MutableLiveData<ArticleDetails>()
+    val downloadedArticleResponse : LiveData<ArticleDetails>
+        get() = _downloadedArticleDetailsResponse
+
+    fun fetchImageDetails(nasaId: String){
+        _networkState.postValue(NetworkState.LOADING)
+
+        try {
+        compositeDisposable.add(
+            apiService.getNasaId(nasaId)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(
+                    {_downloadedArticleDetailsResponse
+                     _networkState.postValue(NetworkState(Status.SUCCESS, "Success $it"))
+                    },
+                    {
+                        _networkState.postValue(NetworkState(Status.FAILED, "Error $it"))
+                    }
+                )
+        )
+    }catch (_: Exception){
```

Comment:
имя параметра не должно быть _

ex: Exception

Внутри должно быть 
```suggestion
    }catch (ex: Exception){ 
    Log.e("NetworkError", ex.printStackTrace())
    }
```